v2/api/apimanagement/v1api20220801/structure.txt

```diff
@@ -0,0 +1,573 @@
+// Code generated by azure-service-operator-codegen. DO NOT EDIT.
+github.com/Azure/azure-service-operator/v2/api/apimanagement/v1api20220801
+├── APIVersion: Enum (1 value)
+│   └── "2022-08-01"
+├── Service: Resource
+│   ├── Owner: github.com/Azure/azure-service-operator/v2/api/resources/v1apiv20191001.ResourceGroup
+│   ├── Spec: Object (24 properties)
+│   │   ├── AdditionalLocations: Object (7 properties)[]
+│   │   │   ├── DisableGateway: *bool
+│   │   │   ├── Location: *string
+│   │   │   ├── NatGatewayState: *Enum (2 values)
+│   │   │   │   ├── "Disabled"
+│   │   │   │   └── "Enabled"
+│   │   │   ├── PublicIpAddressId: *string
+│   │   │   ├── Sku: *Object (2 properties)
+│   │   │   │   ├── Capacity: *int
+│   │   │   │   └── Name: *Enum (6 values)
+│   │   │   │       ├── "Basic"
+│   │   │   │       ├── "Consumption"
+│   │   │   │       ├── "Developer"
+│   │   │   │       ├── "Isolated"
+│   │   │   │       ├── "Premium"
+│   │   │   │       └── "Standard"
+│   │   │   ├── VirtualNetworkConfiguration: *Object (1 property)
+│   │   │   │   └── SubnetResourceId: Validated<*string> (1 rule)
+│   │   │   │       └── Rule 0: Pattern: "^/subscriptions/[^/]*/resourceGroups/[^/]*/providers/Microsoft.(ClassicNetwork|Network)/virtualNetworks/[^/]*/subnets/[^/]*$"
+│   │   │   └── Zones: string[]
+│   │   ├── ApiVersionConstraint: *Object (1 property)
+│   │   │   └── MinApiVersion: *string
+│   │   ├── AzureName: Validated<string> (3 rules)
+│   │   │   ├── Rule 0: MaxLength: 50
+│   │   │   ├── Rule 1: MinLength: 1
+│   │   │   └── Rule 2: Pattern: "^[a-zA-Z](?:[a-zA-Z0-9-]*[a-zA-Z0-9])?$"
+│   │   ├── Certificates: Object (4 properties)[]
+│   │   │   ├── Certificate: *Object (3 properties)
+│   │   │   │   ├── Expiry: *string
+│   │   │   │   ├── Subject: *string
+│   │   │   │   └── Thumbprint: *string
+│   │   │   ├── CertificatePassword: *string
+│   │   │   ├── EncodedCertificate: *string
+│   │   │   └── StoreName: *Enum (2 values)
+│   │   │       ├── "CertificateAuthority"
+│   │   │       └── "Root"
+│   │   ├── CustomProperties: map[string]string
+│   │   ├── DisableGateway: *bool
+│   │   ├── EnableClientCertificate: *bool
+│   │   ├── HostnameConfigurations: Object (11 properties)[]
+│   │   │   ├── Certificate: *Object (3 properties)
+│   │   │   │   ├── Expiry: *string
+│   │   │   │   ├── Subject: *string
+│   │   │   │   └── Thumbprint: *string
```

Comment:
Should these properties on `Certificate` be marked with `$importConfigMapMode: optional` to allow them to be configured after deployment? This partly depends on whether the Cert can be updated after initial deployment - but if it can be updated, making these configurable is a good idea.

Reply:
I'm struggling with adding 
```
      Certificate:
        Expiry:
          $importConfigMapMode: optional
        Subject:
          $importConfigMapMode: optional
        Thumbprint:
          $importConfigMapMode: optional
```
I've tried lots of variants - but if you have some time to chat and guide me - more than happy to add it

Reply:
You want:
```
      CertificateInformation:
        Expiry:
          $importConfigMapMode: optional
        Subject:
          $importConfigMapMode: optional
        Thumbprint:
          $importConfigMapMode: optional
```

Reply:
Darnit! I just came here to make the exact same suggestion. 😁 

In `structure.txt` the name `Certificate` is the name of the referencing property on the _parent_ type; in the config file you need to give the name of the actual type, which isn't shown in `structure.txt`.